### PR TITLE
Small adjusments and bugfixes to 366.

### PR DESCRIPTION
**Pull request by michalpelka**
Instead of a review, I've provided PR. 
Please consider my comments as a review to #366 .

### CODE REVIEW

File: Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp [hunk 1]
```diff
@@ -124,7 +132,7 @@ namespace ROS2
     {
         AZ_TracePrintf("FollowJointTrajectoryActionServer", "Goal accepted\n");
         m_goalHandle = goalHandle;
-        m_goalHandle->execute();
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp:127]: We trigger here denied transition : 
```
terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
  what():  goal_handle attempted invalid transition from state EXECUTING with event EXECUTE, at ./src/rcl_action/goal_handle.c:95
```
It is used only when goals are in the queue.

File: Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp [hunk 2]
```diff
@@ -26,19 +28,33 @@ namespace ROS2
...
             }
-
-            AZ_Printf("JointsManipulationComponent", "Adding joint info for hinge joint %s\n", jointName.GetCStr());
-            JointsManipulationComponent::JointInfo jointInfo;
-            jointInfo.m_isArticulation = false;
-            jointInfo.m_axis = static_cast<PhysX::ArticulationJointAxis>(0);
-            jointInfo.m_entityComponentIdPair = AZ::EntityComponentIdPair(hingeJointComponent->GetEntityId(), hingeJointComponent->GetId());
-            joints[jointName] = jointInfo;
-            return true;
+            // Check if the joint has a PID controller component
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp:37]: Added here some warning when robot's is not configured correctly.

File: Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp [hunk 3]
```diff
@@ -77,30 +94,48 @@ namespace ROS2
...
+            // check if joint has ArticulationJointComponent
+            const AZ::Entity* entity = articulation->GetEntity();
+            auto* articulationJointComponent = Utils::GetGameOrEditorComponent<JointsArticulationControllerComponent>(entity);
+            AZ_Warning(
+                "JointsManipulationComponent",
+                articulationJointComponent,
+                "Joint with name %s (Entity : %s) has no ArticulationJointComponent!",
+                jointName.GetCStr(),
+                entity->GetName().c_str());
+            if (articulationJointComponent)
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp:106]: the same here, warning if ArticulationJointComponent is not added to link.

File: Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp [hunk 4]
```diff
@@ -77,30 +94,48 @@ namespace ROS2
...
+        m_jointStateTopic.m_topic = "joint_states";
+    }
+
     void JointsManipulationComponent::Activate()
     {
         auto* ros2Frame = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(GetEntity());
-
+        m_manipulationJoints.clear();
         JointStatePublisherConfiguration config;
+        config.m_topicConfiguration = m_jointStateTopic;
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp:131]: Bugfix.

File: Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp [hunk 5]
```diff
@@ -324,7 +355,11 @@ namespace ROS2
     {
         if (m_manipulationJoints.empty())
         {
-            return;
+            InitializeJoints();
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/JointsManipulationComponent.cpp:358]: I am not proud of that but works and has no ugly `bool m_initialized` fields.

File: Gems/ROS2/Code/Source/Manipulation/JointsTrajectoryComponent.h [hunk 6]
```diff
@@ -61,7 +61,6 @@ namespace ROS2
         TrajectoryGoal m_trajectoryGoal;
         rclcpp::Time m_trajectoryExecutionStartTime;
 
-        JointsManipulationRequests::ManipulationJoints m_manipulationJoints; //!< Joints: assumes they don't change dynamically.
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/Manipulation/JointsTrajectoryComponent.h:64]: Technically we can dynamically change non-articulated robots, so I would not assume. With implementation from #366 we initialized this field to empty in JointsManipulationRequests::Activation.

File: Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp [hunk 7]
```diff
@@ -81,6 +85,10 @@ namespace ROS2
         if (!IsReadyForExecution())
         {
             AZ_TracePrintf("FollowJointTrajectoryActionServer", "Goal rejected: server is not ready for execution!");
+            if (m_goalHandle)
+            {
+                AZ_TracePrintf("FollowJointTrajectoryActionServer", " m_goalHandle->is_active() = %d m_goalHandle->is_executing() = %d\n", m_goalHandle->is_active(), m_goalHandle->is_executing());
```
Comment (spham-amzn) [inline Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp:90]: The AZ_TracePrintf is being deprecated and should replaced with the AZ_Trace
```suggestion
                AZ_Trace("FollowJointTrajectoryActionServer", " m_goalHandle->is_active() = %d m_goalHandle->is_executing() = %d\n", m_goalHandle->is_active(), m_goalHandle->is_executing());
```

File: Gems/ROS2/Code/Source/Manipulation/JointsTrajectoryComponent.cpp [hunk 8]
```diff
@@ -22,9 +22,6 @@ namespace ROS2
         AZ_Assert(ros2Frame, "Missing Frame Component!");
         AZStd::string namespacedAction = ROS2Names::GetNamespacedName(ros2Frame->GetNamespace(), m_followTrajectoryActionName);
         m_followTrajectoryServer = AZStd::make_unique<FollowJointTrajectoryActionServer>(namespacedAction, GetEntityId());
-
-        JointsManipulationRequestBus::EventResult(m_manipulationJoints, GetEntityId(), &JointsManipulationRequests::GetJoints);
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Manipulation/JointsTrajectoryComponent.cpp:26]: Notification "Joints ready"

File: Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp (hunk 7 above)
Comment (adamdbrw) [reply Gems/ROS2/Code/Source/Manipulation/FollowJointTrajectoryActionServer.cpp]: Thanks for the catch! Will apply in the entire target branch!

**Pull request merged**